### PR TITLE
Update test collection docs

### LINKED ISSUE #2465: Update `Test collection` docs
### Which components does the task require to be changed? (think hard pls)

snforge

### Description

https://foundry-rs.github.io/starknet-foundry/testing/test-collection.html

We should describe that tests are collected from Scarb's [test targets](https://docs.swmansion.com/scarb/docs/reference/targets.html#test-targets). We should keep info that is specific to snforge and its plugin like treating functions tagged with `#[test]` as tests, but remove info that is specific to Scarb test targets (e.g. virtual lib.cairo, explanation of autodetected test targets etc.).

We can have an example for a standard tests structure (tests in both `src/` under `#[cfg(test)]` and in `tests/`) as this is unlikely to change

### PR DESCRIPTION
**Pull request by cptartur**
Closes #2465 

## Introduced changes

- Added information that Scarb handles test collection and linked relevant sections in Scarb docs
- Reworked test collection docs

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: docs/src/testing/test-collection.md [hunk 1]
```diff
@@ -1,95 +1,111 @@
...
-│   ├── test_contract.cairo
-│   └── not_included/
-│       └── ignored.cairo
-└── src/
-    └── lib.cairo
+$ tree
+tests/
+├── module1.cairo <-- is collected
+├── module2.cairo <-- is collected
+└── module3.cairo <-- is collected
```
Comment (franciszekjob) [inline docs/src/testing/test-collection.md:85]: Please split this shell snippet, similarly as in [this PR](https://github.com/foundry-rs/starknet-foundry/pull/2658) (just to maintain consistency 😄 ).

Btw, I think here we should make the output expanded by default.

File: docs/src/testing/test-collection.md [hunk 2]
```diff
@@ -1,95 +1,111 @@
...
 
-tests from `tests/test_contract.cairo`, `tests/common.cairo` and `tests/common/utils.cairo` will be collected.
+```shell
+$ tree
+tests/
+├── lib.cairo
+├── module1.cairo  <-- is collected
+├── module2.cairo  <-- is collected
+└── module3.cairo  <-- is not collected
+```
```
Comment (franciszekjob) [inline docs/src/testing/test-collection.md:104]: Same as above.

File: docs/src/testing/test-collection.md [hunk 3]
```diff
@@ -1,95 +1,111 @@
-## Test Collection
+# How Tests Are Collected
 
-`snforge` considers all functions in your project marked with `#[test]` attribute as tests.
-By default, test functions run without any arguments.
-However, adding any arguments to function signature will enable [fuzz testing](../snforge-advanced-features/fuzz-testing.md) for this
-test case.
```
Comment (kkawula) [inline docs/src/testing/test-collection.md:6]: Do we want to get rid of this information, is it written somewhere else?

File: docs/src/testing/test-collection.md [hunk 4]
```diff
@@ -1,95 +1,111 @@
...
-    └── lib.cairo
-```
+You can read more about tests organization
+in [Scarb documentation](https://docs.swmansion.com/scarb/docs/extensions/testing.html#tests-organization).
+
+### Unit Tests
 
-with `tests/lib.cairo` content:
+Test placed in `src` directory are often called unit tests.
+For these test to function in snforge, they must be wrapped in a module marked with `test` attribute.
```
Comment (ddoktorski) [inline docs/src/testing/test-collection.md:36]:
```suggestion
For these test to function in snforge, they must be wrapped in a module marked with `#[cfg(test)]` attribute.
```

File: docs/src/testing/test-collection.md [hunk 5]
```diff
@@ -1,95 +1,111 @@
...
+tests/
+├── module1.cairo <-- is collected
+├── module2.cairo <-- is collected
+└── module3.cairo <-- is collected
 ```
 
-and `tests/common.cairo` content:
+Scarb will collect each of the file and compile it as a
+separate [test target](https://docs.swmansion.com/scarb/docs/reference/targets.html#test-targets).
+Each of this target will be run separately by `snforge`.
```
Comment (ddoktorski) [inline docs/src/testing/test-collection.md:90]:
```suggestion
Scarb will collect each file and compile it as a
separate [test target](https://docs.swmansion.com/scarb/docs/reference/targets.html#test-targets).
Each of these targets will be run separately by `snforge`.
```

File: docs/src/testing/test-collection.md (hunk 3 above)
Comment (cptartur) [reply docs/src/testing/test-collection.md:6]: We have a separate page on fuzz testing where we explain how to enable them. I don't think the page on test collection benefits from having this explained.

**Review by ddoktorski — APPROVED** [review]

**Review by franciszekjob — APPROVED** [review]

**Review by kkawula — APPROVED** [review]

**Pull request merged**